_posts/2015-07-15-testautomation-allergy.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+title: Test automation allergy
+author: pekkama
+excerpt: Questioning the snake-oil approach to test automation
+---
+
+I am allergic to test automation!
+---------------------------------
+
+That's correct! I get itchy and my eyes redden every time some talks about test automation. I get coarse throat when I see a test automation coverage report. I get twitches when we justify our lack of skills as testers with 95% line coverage. That's a reaction to poorly executed and poorly justified test automation. I am allergic to s***ty test automation. And I think everyone should be!
+
+This post is about questioning the snake-oil approach to test automation. I have been doing software testing for a number of years in different types of organizations and workplaces. My approach has been skill-driven and holistic, and it is still today. Test automation done well is worth the money invested in it. There has been only few cases where it has been worth the money, though. The best automation is always a supporting act to other testing activities, and the best automation is treated that way.
+
+
+You cannot automate testing!
+----------------------------
+
+Testing isn't a set of steps. By trivializing testing to just execution of steps is plain stupidity! Everyone who thinks like that is essentially in denial, ignorant or a stupid in the head. Testing is a vast set of mental and physical activities. It is executing tests, designing tests, analyzing the results, guessing, trying, learning, understanding, and reporting. More than anything, testing is thinking. Machines don't think, they assert only the stuff we tell them to. We cannot create so comprehensive, complex machine that does the job of a human brain. We can automate something, yes. My suggestion is to understand the limits of test automation and make the best of the stuff it is good at.
```

Comment:
"By trivializing testing to just execution of steps is plain stupidity!" => "Trivializing testing to just execution of steps is plain studidity!"